htap.go

```diff
@@ -22,5 +25,33 @@ func (qcc *queryContextCache) init() *queryContextCache {
 func (qcc *queryContextCache) add(qces ...queryContextEntry) {
 	qcc.mutex.Lock()
 	defer qcc.mutex.Unlock()
-	qcc.entries = append(qcc.entries, qces...)
+	var newQcesBasedOnPriorities []queryContextEntry
+	for _, newQce := range qces {
+		logger.Debugf("adding query context: %v", newQce)
+		newQceProcessed := false
+		for existingQceIdx, existingQce := range qcc.entries {
+			if newQce.ID == existingQce.ID {
+				newQceProcessed = true
```

Comment:
we want to keep it true when newQce.Timestamp is lower than the existing one?

Reply:
Yes. This is because here we considered it and decided that we don't use that qce. If we don't it would be added either way at the end of the loop.